examples/applications/plot_time_series_lagged_features.py

```diff
@@ -1,425 +1,433 @@
-# series evaluation strategy using
-# :class:`~sklearn.model_selection.TimeSeriesSplit`. We observed that the
-# model trained using :class:`~sklearn.model_selection.train_test_split`,
-# having a default value of `shuffle` set to `True` produced an overly
-# optimistic Mean Average Percentage Error (MAPE). The results
-# produced from the time-based split better represent the performance
-# of our time-series regression model. We also analyzed the predictive uncertainty
-# of our model via Quantile Regression. Predictions based on the 5th and
-# 95th percentile using `loss="quantile"` provide us with a quantitative estimate
-# of the uncertainty of the forecasts made by our time series regression model.
-# Uncertainty estimation can also be performed
-# using `MAPIE <https://mapie.readthedocs.io/en/latest/index.html>`_,
-# that provides an implementation based on recent work on conformal prediction
-# methods and estimates both aleatoric and epistemic uncertainty at the same time.
-# Furthermore, functionalities provided
-# by `sktime <https://www.sktime.net/en/latest/users.html>`_
-# can be used to extend scikit-learn estimators by making use of recursive time
-# series forecasting, that enables dynamic predictions of future values.
+"""
+===========================================
+Lagged features for time series forecasting
+===========================================
+
+This example demonstrates how Polars-engineered lagged features can be used
+for time series forecasting with
+:class:`~sklearn.ensemble.HistGradientBoostingRegressor` on the Bike Sharing
+Demand dataset.
+
+See the example on
+:ref:`sphx_glr_auto_examples_applications_plot_cyclical_feature_engineering.py`
+for some data exploration on this dataset and a demo on periodic feature
+engineering.
+
+"""
+
+# %%
+# Analyzing the Bike Sharing Demand dataset
+# -----------------------------------------
+#
+# We start by loading the data from the OpenML repository as a raw parquet file
+# to illustrate how to work with an arbitrary parquet file instead of hiding this
+# step in a convenience too such as `sklearn.datasets.fetch_openml`.
+#
+# The URL of the parquet file can be found in the JSON description of the
+# Bike Sharing Demand v7 dataset on openml.org.
+import numpy as np
+import polars as pl
+
+from sklearn.datasets import fetch_file
+
+pl.Config.set_fmt_str_lengths(20)
+
+bike_sharing_data_file = fetch_file(
+    "https://openml1.win.tue.nl/datasets/0004/44063/dataset_44063.pq",
+    sha256="d120af76829af0d256338dc6dd4be5df4fd1f35bf3a283cab66a51c1c6abd06a",
+)
+bike_sharing_data_file
+
+# %%
+# We load the parquet file with Polars for feature engineering. Polas
+# automatically caches common subexpressions which are reused in multiple
+# expressions (like `pl.col("count").shift(1)` below). See
+# https://docs.pola.rs/user-guide/lazy/optimizations/ for more information.
+
+df = pl.read_parquet(bike_sharing_data_file)
```

Comment:
Note to reviewers, for some reason this file had Windows-style CRLF line endings. This PR only changes the content of first 2 cells.